doc/_scripts/software_maturity_scanner.py

```diff
@@ -0,0 +1,554 @@
+a 'software_maturity_features.yaml' file.
+
+The ``TWISTER_OUT_DIR`` environment variable is used by default to provide the
+necessary input folder. It can be overwritten by supplying the
+``--build-output`` option.
+
+The Azure access key can be supplied with the argument ``--access-key``.
+If not provided as an argument explicitly, the environment variable
+``AZURE_REPORT_CACHE_ACCESS_KEY`` is used by default.
+
+Publish Results/Run Locally
+***************************
+The script can run in two different modes, based on which of the following
+options are supplied: ``--save-output`` or ``--publish-results``.
+
+When ``--save-output`` is used, no Azure access keys are needed, and the
+results are written to the specified file on the local filesystem.
+
+When run with the ``--publish-results``, the script will run as normal, and
+upload the resulting YAML file to Azure blob storage. This should ONLY be
+done by the CI build system. To differentiate the results of different runs,
+use the ``--commit-sha`` option upon execution to mark the commit on which the
+build output is based.
+
+Script Logic
+************
+
+A complete list of all available SoCs are found by searching through the
+samples and applications folder for sample.yaml files and retrieving the
+``integration_platforms`` list from each. Anything nRF51 and not starting with
+nRF is excluded.
+
+To mark features as experimental the twister output folder is searched for
+all build.log files and scanned for the string "Experimental symbol XXX is
+enabled". When found, The SoC is looked up in /zephyr/.config, and a mapping
+from each SoC to the experimental symbols detected when using it to build is
+stored and used to evaluate the Kconfig rules.
+
+Next, the twister output folder is searched for all .config-files. These
+are evaluated against the given Kconfig rules to determine which features
+are supported, experimental or not supported. If a rule contains multiple
+symbols connected with either && or ||, all of the symbols must be present in
+the same .config-file for a SoC to be registered as supporting that feature.
+That means that a feature is not marked as supported if there is no explicit
+test case where all the required Kconfig symbols are present at the same time,
+even if the required options have been enabled individually in separate
+samples. If at least one of the symbols are marked as experimental, the feature
+is also marked as experimental.
+
+In cases where there are rules with multiple symbols connected with ||, and one
+is experimental and the others are not, the feature will still be marked as
+experimental. If this is an undesired outcome, the feature should be split
+into two separate features, one with the experimental symbols and the other
+with the supported ones.
+
+Copyright (c) 2022 Nordic Semiconductor ASA
+"""
+
+from pathlib import Path
+from typing import Dict, List, Set
+from azure.storage.blob import ContainerClient
```

Comment:
do we have any file which contains pip requirements for this file?

Reply:
The script seems to work in a fresh virtual environment with python 3.8.10 given these requirements:
```
azure-storage-blob==12.9.0
kconfiglib==14.1.0
PyYAML==5.4.1
```

Should this be stored in a specific file somewhere?